flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterTest.java

```diff
@@ -2097,6 +2098,87 @@ public void testResourceRequirementsAreRequestedFromTheScheduler() throws Except
         }
     }
 
+    @Test
+    public void testJobMasterDoesNotReconnectToResourceManagerEvenIfCleanupStalls()
+            throws Exception {
+
+        // the ResourceManager should count the connect attempts
+        final TestingResourceManagerGateway resourceManagerGateway =
+                createAndRegisterTestingResourceManagerGateway();
+        final AtomicInteger connectCount = new AtomicInteger();
+        final OneShotLatch firstRegistrationLatch = new OneShotLatch();
+        resourceManagerGateway.setRegisterJobManagerFunction(
+                (jobMasterId, resourceID, s, jobID) -> {
+                    connectCount.incrementAndGet();
+                    firstRegistrationLatch.trigger();
+
+                    return CompletableFuture.completedFuture(
+                            resourceManagerGateway.getJobMasterRegistrationSuccess());
+                });
+
+        // the scheduler close logic should be handled outside the main thread
```

Comment:
You're right - the comment is not really helpful because the intention of the future is documented further down in the test method where the `schedulerCloseFuture` is actually completed.